main/src/ca/uwaterloo/flix/util/CofiniteSet.scala

```diff
@@ -52,6 +53,11 @@ sealed trait CofiniteSet[T] {
 
 object CofiniteSet {
 
+  /**
+    * A trait that indicates the empty and universal values of this type have been cached.
+    *
+    * We use this to ensure there is ONE empty and ONE universe value (per type) throughout the program.
```

Comment:
I does not really explain why this is important. Is it for semantics or just performance??

Reply:
Also how does it guarantee it